diff --git a/core/lib/Drupal/Core/Block/BlockPluginInterface.php b/core/lib/Drupal/Core/Block/BlockPluginInterface.php
index e4b330517b..f00baf6bb4 100644
--- a/core/lib/Drupal/Core/Block/BlockPluginInterface.php
+++ b/core/lib/Drupal/Core/Block/BlockPluginInterface.php
@@ -149,8 +149,8 @@ public function blockSubmit($form, FormStateInterface $form_state);
    * is only responsible for providing a baseline suggestion; calling code is
    * responsible for ensuring whatever uniqueness is required for the use case.
    *
-   * @return string
-   *   The suggested machine name.
+   * @return string|null
+   *   The suggested machine name or NULL if not applicable.
    */
   public function getMachineNameSuggestion();
 
diff --git a/core/lib/Drupal/Core/Block/BlockPluginTrait.php b/core/lib/Drupal/Core/Block/BlockPluginTrait.php
index acb850ed01..07bd9b122c 100644
--- a/core/lib/Drupal/Core/Block/BlockPluginTrait.php
+++ b/core/lib/Drupal/Core/Block/BlockPluginTrait.php
@@ -246,6 +246,10 @@ public function blockSubmit($form, FormStateInterface $form_state) {
    */
   public function getMachineNameSuggestion() {
     $definition = $this->getPluginDefinition();
+    if (!isset($definition['admin_label'])) {
+      return NULL;
+    }
+
     $admin_label = $definition['admin_label'];
 
     // @todo This is basically the same as what is done in
diff --git a/core/modules/block/src/BlockForm.php b/core/modules/block/src/BlockForm.php
index 90cd6823ff..671808264e 100644
--- a/core/modules/block/src/BlockForm.php
+++ b/core/modules/block/src/BlockForm.php
@@ -382,11 +382,15 @@ protected function submitVisibility(array $form, FormStateInterface $form_state)
    * @param \Drupal\block\BlockInterface $block
    *   The block entity.
    *
-   * @return string
+   * @return string|null
    *   Returns the unique name.
    */
   public function getUniqueMachineName(BlockInterface $block) {
     $suggestion = $block->getPlugin()->getMachineNameSuggestion();
+    if (!$suggestion) {
+      return NULL;
+    }
+
     if ($block->getTheme()) {
       $suggestion = $block->getTheme() . '_' . $suggestion;
     }
diff --git a/core/tests/Drupal/Tests/Core/Block/BlockBaseTest.php b/core/tests/Drupal/Tests/Core/Block/BlockBaseTest.php
index 2b9cd98b84..f0c329aebc 100644
--- a/core/tests/Drupal/Tests/Core/Block/BlockBaseTest.php
+++ b/core/tests/Drupal/Tests/Core/Block/BlockBaseTest.php
@@ -40,6 +40,46 @@ public function testGetMachineNameSuggestion($label, $expected) {
     $this->assertEquals($expected, $block_base->getMachineNameSuggestion());
   }
 
+  /**
+   * Tests the machine name suggestion when no label is available.
+   *
+   * @see \Drupal\Core\Block\BlockBase::getMachineNameSuggestion()
+   */
+  public function testGetMachineNameSuggestionEmptyLabel() {
+    $previous_error_handler = set_error_handler(function ($severity, $message, $file, $line) use (&$previous_error_handler) {
+      // Convert deprecation error into a catchable exception.
+      if ($severity === E_DEPRECATED) {
+        throw new \ErrorException($message, 0, $severity, $file, $line);
+      }
+      if ($previous_error_handler) {
+        return $previous_error_handler($severity, $message, $file, $line);
+      }
+    });
+
+    $module_handler = $this->createMock('Drupal\Core\Extension\ModuleHandlerInterface');
+    $transliteration = $this->getMockBuilder('Drupal\Core\Transliteration\PhpTransliteration')
+      ->setConstructorArgs([NULL, $module_handler])
+      ->onlyMethods(['readLanguageOverrides'])
+      ->getMock();
+
+    $config = [];
+    $definition = [
+      'admin_label' => NULL,
+      'provider' => 'block_test',
+    ];
+    $block_base = new TestBlockInstantiation($config, 'test_block_instantiation', $definition);
+    $block_base->setTransliteration($transliteration);
+
+    try {
+      $this->assertEquals(NULL, $block_base->getMachineNameSuggestion());
+    }
+    catch (\ErrorException $e) {
+      $this->fail('Exception thrown when no label provided.');
+    }
+
+    restore_error_handler();
+  }
+
   /**
    * Provides data for testGetMachineNameSuggestion().
    */
@@ -47,6 +87,7 @@ public function providerTestGetMachineNameSuggestion() {
     return [
       ['Admin label', 'adminlabel'],
       ['über åwesome', 'uberawesome'],
+      [NULL, NULL]
     ];
   }
 
